From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marcus Hall <marcus@tuells.org>
Date: Mon, 18 Jan 2021 14:53:11 -0700
Subject: [PATCH] 2662: CreateFunctionCmd -- Return called addr instead of null

Code is trying to find the proper entry point and has found an instruction that is the To address of a call reference.  We should return the address of this instruction.  The original code would return fallFrom which was found to be null on line 573 above it.
---
 .../main/java/ghidra/app/cmd/function/CreateFunctionCmd.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java b/Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java
index 122873e00..b919fb559 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/cmd/function/CreateFunctionCmd.java
@@ -575,7 +575,7 @@ public class CreateFunctionCmd extends BackgroundCommand {
 				}
 				Reference ref = iter.next();
 				if (ref.getReferenceType().isCall()) {
-					entry = fallFrom;
+					entry = followInstr.getAddress();
 					break;
 				}
 				fallFrom = ref.getFromAddress();
-- 
2.40.1

